src/webcomponents/helperfunctions/jsonToCol.ts

```diff
@@ -0,0 +1,124 @@
+export const jsonToCol = (languages: string): any => {
+  let datas: any = JSON.parse(languages);
```

Comment:
do not parse json here - input parameter should by type of `LanguagesInput` that you defined in another file (just import it here)

Reply:
map doesn't work on a string. that's why it needs to be parsed to json.

Reply:
<iteria-messages-spreadsheet languages="{
   'en': {'budget': 'BUDGET', 'message2': 'MESSAGE2'}, 
   'sk': {'budget': 'ROZPOCET', 'message2': 'hlaska 2'}
}" onchange="console.log(event)">
</iteria-messages-spreadsheet>

there YOU used strings. That's why I used String

Reply:
I will be more explicit. Parse JSON before calling `jsonToCol()`. 
Use following code as inspiration.
```
const inLanguages = JSON.parse(languages);
jsonToCol(inLanguages)
```

```
export const jsonToCol = (languages: LanguagesInput) => {
   // no unnecessary local variable (use cont)
   const columns = mapLangToSheetColumns(languages)
   const data = mapLangToSheetData(languages)
  return { columns, data }
}
```

Reply:
Thank you very much for the suggestion. mapLangToSheetColumns and mapLangToSheetData is this 2 function?

Reply:
Sure